_episodes/35.md

```diff
@@ -0,0 +1,240 @@
+done manually or with many existing tools.
+
+However you can also take it a step further and create your own package suited
+to your needs. The tarball can be used as source for any customization to create
+your own package. In the following is a list of options and scenarios:
+
+* Use the tarball, but remove unused plugins.
+* Use the tarball as source to create your own specific package. For example a
+  deb archive for usage with Ubuntu, or an Alpine package for that same distro.
+* Create your own RPM similar to [Manfred's proof of
+  concept](https://github.com/simpligility/trino-packages) that pulls out the
+  Trino RPM package creation into a separate project.
+* Create your own container image with different base distro, custom set of
+  plugins, and even with all your configuration baked into the image.
+
+### Others
+
+You can also use brew on macos, but that is not suitable for production usage.
+More for convenience to get a local Trino for playing around.
+
+## Additional topic of the episode: Modernizing Trino with Java 17
+
+Currently Java 11 is required for Trino. Java 17 is the latest and greatest Java
+LTS release with lots of good performance, security, and language improvements.
+The community has been working hard to make Java 17 support a reality. At this
+stage core Trino fully supports Java 17. Starburst Galaxy for example uses Java
+17.
+
+The maintainers and contributors would like to move to fully support and also
+require Java 17 soon. And here is where your input comes in, and we ask that you
+let us know your thoughts about questions such as the following:
+
+* Are you looking forward to the new Java 17 language features and other
+  improvements as a contributor to Trino?
+* Are you already using Java 17 with Trino? In production or just testing?
+* If we require Java 17 in the next months, can you update to use Java 17 with
+  Trino?
+* If not, what are some of the hurdles?
+* Are you okay with staying at an older release, until you can use Java 17?
+
+Let us know on the #dev channel on Trino Slack or ping us directly. You can also
+chime in on the [roadmap issue](https://github.com/trinodb/trino/issues/9876).
+
+
+## Pull requests of the episode: Worker stats in the Web UItbd
+
+The [PR of the episode](https://github.com/trinodb/trino/issues/11653) was
+submitted [Github user whutpencil](https://github.com/whutpencil) and adds a
+significant new feature to the web UI. It exposes the system.runtimes.nodes
+information, so statistics for each worker, in brand new pages. What a great
+effort! Special thanks also go out to [Dawid Adamek
+`dedep`](https://github.com/dedep) for the review.
+
+## Demo of the episode: Tarball installation and new Web UI feature
+
+In the demo of the month Manfred shows a worker installation to add to a local
+tarball install of a coordinator, and then demos the Web UI with the new feature
+from the pull request of the month.
+
+## Question of the episode: tbd
+
```

Comment:
Gonna cite this question:

Title: Are write operations in delta lake supported for tables stored on HDFS?

Full question: I was trying the delta lake connector. I noticed that write operations are supported for tables stored on Azure ADLS Gen2, S3 and S3-compatible storage. Does that mean write operations are not supported for tables stored on HDFS?

Slack Link: https://trinodb.slack.com/archives/CGB0QHWSW/p1650331073409229

Answer: HDFS is always implicitly supported for data lake connectors. It isn’t called out because it is assumed.

The confusion actually came from an error message used when this user tried to insert into a delta lake table they created in Spark

Then They tried inserting a record into the table through IntelliJ IDEA. When I clicked the submit button, I got error message below:
Unsupported target SQL type: -155
They thought the problem might be the wrong data type of birthday. Then used statement below to insert a record into the table.

INSERT INTO presto.people10m (id, firstname, middlename, lastname, gender, birthdate, ssn, salary) VALUES (1, 'a', 'b', 'c', 'male', timestamp '1990-01-01 00:00:00 +00:00', 'd', 10);

However, I got an error message like this:

Query 20220419_031201_00015_8qe76 failed: Cannot write to table in hdfs://masters/presto.db/people10m; hdfs not supported

This was an issue on the IntelliJ client